api/src/main/java/org/pmiops/workbench/notebooks/NotebooksApiClientTracer.java

```diff
@@ -0,0 +1,38 @@
+package org.pmiops.workbench.notebooks;
+
+import com.squareup.okhttp.Call;
+import com.squareup.okhttp.Response;
+import io.opencensus.common.Scope;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+/** Created by brubenst on 7/22/19. */
+public class NotebooksApiClientTracer extends ApiClient {
+  private static final Tracer tracer = Tracing.getTracer();
+
+  @Override
+  public <T> ApiResponse<T> execute(Call call, Type returnType) throws ApiException {
+    Response response;
+    T data;
+    try (Scope ss =
+        tracer
+            .spanBuilderWithExplicitParent("NotebooksApiCall", tracer.getCurrentSpan())
+            .startScopedSpan()) {
+      response = call.execute();
```

Comment:
Can you just replace all this logic with `return super.execute()`? Wrapped with the session block?

Reply:
I can't, because I need to manually open and close the trace for the URL.